src/components/Modal.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import '../styles/Modal.css';
+
+export default class Modal extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: true
+    };
+  }
+
+  closeModal = () => {
+    const { isOpen } = this.state;
+    if (isOpen === true) {
+      this.setState({ isOpen: false });
+    }
+  };
+
+  render() {
+    const { header, body, action } = this.props;
+    const { isOpen } = this.state;
+    return (
+      <div className={`modal ${isOpen ? '' : 'close'}`}>
+        <div className="modal-content">
+          <div className="modal-header">{header}</div>
+          <div className="modal-body">{body}</div>
+          <button
+            onClick={this.closeModal}
```

Comment:
My suggestion here would be to have this component take in a function as a prop. Let's say that prop is called actionFunction, then have this be something like: 
```suggestion
            onClick={ action || this.closeModal}
```

Reply:
basically, to have the default behavior of this prop be to close the modal, but to allow the component to have this button to have  additional functionality